prqlc/prql-compiler/src/sql/gen_expr.rs

```diff
@@ -104,6 +104,32 @@ pub(super) fn translate_expr(expr: Expr, ctx: &mut Context) -> Result<ExprOrSour
                 }
                 "std.concat" => return Ok(process_concat(&expr, ctx)?.into()),
                 "std.array_in" => return Ok(process_array_in(args, ctx)?.into()),
+                "std.date.to_string" => {
+                    if let [date_format_exp @ Expr {
+                        kind: ExprKind::Literal(Literal::String(date_format)),
+                        ..
+                    }, col_expr] = &args[..]
+                    {
+                        // Translate PRQL date format to dialect specific date format
+                        // and then continue with the standard approach.
+                        let expr = Expr {
+                            kind: ExprKind::Operator {
+                                name: name.to_string(),
+                                args: vec![
+                                    Expr {
+                                        kind: ExprKind::Literal(Literal::String(
+                                            ctx.dialect.translate_prql_format(date_format),
+                                        )),
+                                        span: date_format_exp.span,
+                                    },
+                                    col_expr.clone(),
+                                ],
+                            },
+                            ..expr
+                        };
+                        return super::operators::translate_operator_expr(expr, ctx);
+                    }
```

Comment:
Add an else branch that produces an error saying "function std.date.to_string, param format expected a string literal".

Also add the following test:
```
from dates_to_display
select {my_date, my_format}
select {std.date.to_string my_date my_format}
```